app/src/play/java/chat/rocket/android/dynamiclinks/DynamicLinksForFirebase.kt

```diff
@@ -0,0 +1,58 @@
+package chat.rocket.android.dynamiclinks
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import chat.rocket.android.util.TimberLogger
+import chat.rocket.android.R
+import com.google.firebase.dynamiclinks.DynamicLink
+import com.google.firebase.dynamiclinks.FirebaseDynamicLinks
+import com.google.firebase.dynamiclinks.ShortDynamicLink
+import javax.inject.Inject
+
+// DEBUG
+import android.widget.Toast
+
+class DynamicLinksForFirebase @Inject constructor(private var context: Context) :
+        DynamicLinks {
+
+    private var deepLink: Uri? = null
+    private var newDeepLink: String? = null
+
+    override fun getDynamicLink(intent: Intent, deepLinkCallback: (Uri?) -> Unit?) {
+
+        FirebaseDynamicLinks.getInstance()
+                .getDynamicLink(intent)
+                .addOnSuccessListener { pendingDynamicLinkData ->
+                    if (pendingDynamicLinkData != null) {
+                        deepLink = pendingDynamicLinkData.link
+                    }
+                    deepLinkCallback(deepLink)
+                }
+                .addOnFailureListener { e -> TimberLogger.debug("getDynamicLink:onFailure : $e") }
+    }
+
+    override fun createDynamicLink(username: String, server: String, deepLinkCallback: (String?) -> Unit? ) {
+
+        FirebaseDynamicLinks.getInstance().createDynamicLink()
+                .setLink(Uri.parse("$server/direct/$username"))
+                .setDomainUriPrefix("https://" + context.getString(R.string.dynamiclink_host))
+                .setAndroidParameters(
+                        DynamicLink.AndroidParameters.Builder(context.getString(R.string.app_package_name)).build())
+                .setSocialMetaTagParameters(
+                        DynamicLink.SocialMetaTagParameters.Builder()
+                                .setTitle(username)
+                                .setDescription("Chat with $username on " + server)
+                                .build())
+                .buildShortDynamicLink(ShortDynamicLink.Suffix.SHORT)
+                .addOnSuccessListener { result ->
+                    newDeepLink = result.shortLink.toString()
+                    Toast.makeText(context, newDeepLink, Toast.LENGTH_SHORT).show()
+                    deepLinkCallback(newDeepLink)
+
+                }.addOnFailureListener {
+                    // Error
+                    Toast.makeText(context, "Error dynamic link", Toast.LENGTH_SHORT).show()
```

Comment:
These toast messages were just for debugging.  I will remove them and replace with timber logging without string translations. If you would rather we do it a different way please let me know.

Reply:
Timber is enough but if it is there because you are still debugging it I'll make this as a WIP PR, ok?

Anyway, I think that showing an error message to the user when there is a failure would be better. In this case make sure to implement the `MessageView` on the fragment (if needed) and make use of our ext. function `showToast(resId)` and/or `showToast(message)` plus log it with timber.

Reply:
Is it supposed to show the deeplink url on the toast in this line (on the `addOnSuccessListener`):
`Toast.makeText(context, newDeepLink, Toast.LENGTH_SHORT).show()`
?

 Only a toast with the deeplink url? 🤔 
If so, better to add more context on the message.

Reply:
My thought is this.... I don't think we want to show any info to the user about the deepLink.  Users have no idea what a deepLink is, and there is no requirement that they do.  I added the logging with Timber just to have a record in the server logs for admins as needed.  If the deepLink creation fails, then it uses the default message, and the user has not been affected.  Thoughts?  

FYI - I did push a bunch of updates to the PR and the Toast messages are now removed.